javascript/ql/test/experimental/PoI/CommandInjectionPoIConfiguration.ql

```diff
@@ -0,0 +1,15 @@
+/**
+ * @kind problem
+ */
+
+import javascript
+import experimental.PoI.PoI::PoI
+import semmle.javascript.security.dataflow.CommandInjection
+import semmle.javascript.security.dataflow.IndirectCommandInjection
+import semmle.javascript.security.dataflow.ShellCommandInjectionFromEnvironment
+
+class Config extends StandardPoIConfigurations::DataFlowConfigurationPoIConfiguration {
+  Config() { this = "Config" }
```

Comment:
Are we likely to have multiple subclasses of these standard PoI configurations?

Because it seems to me we can just move the charpred into the base class, remove the bindingset, but let it remain abstract. That way, simply having a non-abstract subclass is all it takes to activate it.

Reply:
Good point, I will implement the suggested change. 
One benefit of the current architecure is that we can detect if someone accidentally activates two configurations at once since `count(PoIConfiguration c) = 2`. But I would rather save the redundant constructor, and debug the rare case manually.

For reference, I did play around with having another inheritance level such that  the following was handled by a `DefaultPoIConfiguration`, but that is too messy to be worth it IMO.
```
      not exists(PoIConfiguration cfg) and
      PoIConfigDefaults::enabled(poi) and
      PoIConfigDefaults::shown(l1) and
      PoIConfigDefaults::shown(l2) and
      PoIConfigDefaults::shown(l3)
```

Reply:
Hmm. Although it is much more compact, it results in a compiler warning:
 
```ql
class Config extends PoIConfiguration {
  // ^^^^^^ WARNING: Config extends only abstract classes but does not provide its own characteristic predicate
  override predicate enabled(PoI poi) { poi instanceof UnpromotedRouteHandlerPoI }
}
```

Reply:
Oh, right. Well let's just stick with the original design then.

Reply:
I got a much better API now. 

Essence:

1.
I have replaced the `StandardPoIConfigurations` with inclusive subclasses instead:
```ql
   class ServerRelatedPoI extends PoI {
      ServerRelatedPoI() {
        this instanceof UnpromotedRouteSetupPoI or
        this instanceof UnpromotedRouteHandlerPoI or
        this instanceof UnpromotedRouteHandlerWithFlowPoI
      }
    }
```
This is much more flexible and easier to compose.

2.
`DefaultEnabledPoI` has been introduced to avoid the awkward `not poi instanceof StandardPoI`. This enables the one-liners for reimplementing what was previously done with a `PoIConfiguration`:

```ql
class MyServerRelatedPoI extends ServerRelatedPoI, DefaultEnabledPoI {}
```
instead of
```ql

class Config extends PoIConfiguration {
  Config() { this = "Config" }

  override predicate enabled(PoI poi) { poi instanceof ServerRelatedPoI }
}
```